src/makeCalculator.js

```diff
@@ -37,7 +37,51 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(value) {
+      this.result += value;
+
+      return this;
+    },
+    subtract(value) {
+      this.result -= value;
+
+      return this;
+    },
+    multiply(value) {
+      this.result *= value;
+
+      return this;
+    },
+    divide(value) {
+      if (value > 0) {
+        this.result /= value;
+
+        return this;
+      }
+    },
+    validateValue(value) {
+      return typeof value === 'number' && !isNaN(value);
+    },
+    validateOperation(operation) {
+      return typeof operation === 'function';
+    },
+    reset() {
+      if (this.validateOperation(this.reset)) {
+        this.result = 0;
+
+        return this;
+      }
+    },
+    operate(operation, value) {
+      if (this.validateOperation(operation) && this.validateValue(value)) {
+        return operation.call(this, value);
```

Comment:
You're not handling situations where the operation is not valid or the value is not valid. It would be better to throw an error or return a specific value (like null or false) in these cases or you just remove this condition